### DISCUSSION TITLE
MaxLFQ Ignoring Missing Values in diann_maxlfq (R Package)

### DISCUSSION THREAD

**Discussion by rlumpkin** [vdemichev/DiaNN#746] [Q&A]
How does the MaxLFQ function in the DiaNN R package deal with mixed missing and non-missing values? It seems like it completely ignores missing values for a given sample if a non-zero value is present. 

Is there any way to avoid this behavior? I have a protein with 30 peptides and a sample with missing values in 29/30 of the peptides, which I would consider to be very meaningfully missing, yet MaxLFQ assigns it a value roughly equal to the one non-missing value.

Thank you!

**Comment by Mickey9292** [comment]
Hi rlumpin!
I am very interesting about that topic.
We’ve also seen this phenomenon in the pg.matrix report. We’ve tried the last version of diaNN 1.8.2 beta 22, with the new quantification options (precision and accuracy) and the results are the same. Some KO proteins are detected present using diaNN just because of one peptide over 10-20.
Hope Vadim will help us to understand.

**Reply by vdemichev** [reply]
Yes, this will happen because of lack of run-specific protein filtering applied to pg_matrix by default - this is OK for many application and is currently the most common preference. Using the main report & filtering based on PG.Q.Value should address this.

Best,
Vadim

**Comment by haiyan-MS** [comment]
I have the same problem. So eventrually I decided to use the full report and some filtering to get satisfactory results. however, if we do that, MaxLFQ is not proporate and we have to use normalized intensity.
